src/wp-includes/block-supports/duotone.php

```diff
@@ -538,3 +591,21 @@ function wp_render_duotone_support( $block_content, $block ) {
 	)
 );
 add_filter( 'render_block', 'wp_render_duotone_support', 10, 2 );
+
+/**
+ * Render the SVG filters supplied by theme.json.
+ *
+ * Note that this doesn't render the per-block user-defined
+ * filters which are handled by wp_render_duotone_support,
+ * but it should be rendered in the same location as those to satisfy
+ * Safari's rendering quirks.
+ *
+ * @since 5.9.0
+ */
+function wp_global_styles_render_svg_filters() {
+	$filters = wp_get_global_styles_svg_filters();
+	if ( ! empty( $filters ) ) {
+		echo $filters;
+	}
+}
+add_action( 'wp_body_open', 'wp_global_styles_render_svg_filters' );
```

Comment:
For some reason the SVG filters are not rendering in the post-editor for me as if `wp_body_open()` is not being called. It works fine when viewing the post. Testing with TwentyTwentyTwo with the following added to the theme.json.

```json
{
    "styles": {
        "blocks": {
            "core/image": {
                "filter": {
                    "duotone": "var(--wp--preset--duotone--primary-and-secondary)"
                }
            }
        }
    }
}
```

Reply:
The [`wp_body_open`](https://developer.wordpress.org/reference/functions/wp_body_open/) is template action. I don't think it's called or should be called in Dashboard.

Reply:
Hmm. It was being called with this setup back when I wrote that code. We'll have to find a similar action that gets called in `<body>` before everything else, then